### PR TITLE
Resource files for mac should be copied to the app bundle

### PR DESCRIPTION
**Pull request by timrisi**
For https://bugzilla.xamarin.com/show_bug.cgi?id=45696

Resource files are compiled into the bin folder but not copied into the actual app folder currently.  This copies them to where they need to be.

### CODE REVIEW

File: tools/mmp/Assembly.cs [hunk 1]
```diff
@@ -21,5 +25,52 @@ bool CopyAssembly (string source, string target)
...
+
+			foreach (var a in Satellites) {
+				string target_dir = Path.Combine (directory, Path.GetFileName (Path.GetDirectoryName (a)));
+				string target_s = Path.Combine (target_dir, Path.GetFileName (a));
+
+				if (!Directory.Exists (target_dir))
+					Directory.CreateDirectory (target_dir);
+
+				CopyAssembly (a, target_s);
+			}
```
Comment (spouliot) [inline tools/mmp/Assembly.cs]: That's a copy of the same method in `tools/common/Assembly.cs`. The code should be moved into `tools/common/Assembly.cs` which is where the shared mtouch/mmp code aside.

**Comment by timrisi** [comment]
Build

File: tools/mmp/driver.cs [hunk 2]
```diff
@@ -1700,6 +1700,8 @@ static void CopyI18nAssemblies (I18nAssemblies i18n)
 
 				// The linker later gets angry if you copy in a read only assembly
 				CopyFileAndRemoveReadOnly (asm, Path.Combine (mmp_dir, filename));
+				foreach (var assembly in BuildTarget.Assemblies)
```
Comment (rolfbjarne) [inline tools/mmp/driver.cs]: You're iterating over all the assemblies inside a foreach of all the assemblies. Shouldn't this new foreach be placed outside the containing foreach?

**Comment by rolfbjarne** [comment]
monotouch-test has tests for this, but iirc they were not ported to XM. It would be nice to get those tests working on XM as well.

**Comment by chamons** [comment]
@timrisi Can you merge in master and see what we need to get this in?

**Comment by timrisi** [comment]
Failures unrelated (media player made 64-bit, handled in a separate PR)

**Pull request merged**